src/components/calculator.js

```diff
@@ -1,33 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Button from './Buttons';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const [value, setValue] = useState({ total: '0', next: null, operation: null });
+
+  function handleClick(digit) {
+    setValue(calculate(value, digit));
+  }
+
+  const { total, next, operation } = value;
+
   return (
     <div className="calculator-grid">
       <div className="output">
         <div className="current-result">
-          0
+          {total}
+          {operation}
+          {next}
         </div>
       </div>
-      <Button value="AC" />
-      <Button value="+/-" />
-      <Button value="%" />
-      <Button value="÷" bg="orange" />
-      <Button value={7} />
-      <Button value={8} />
-      <Button value={9} />
-      <Button value="*" bg="orange" />
-      <Button value={4} />
-      <Button value={5} />
-      <Button value={6} />
-      <Button value="-" bg="orange" />
-      <Button value={1} />
-      <Button value={2} />
-      <Button value={3} />
-      <Button value="+" bg="orange" />
-      <Button value={0} span="span1" />
-      <Button value="." />
-      <Button value="=" bg="orange" />
+      <Button value="AC" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="+/-" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="%" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="÷" bg="orange" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="7" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="8" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="9" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="*" bg="orange" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="4" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="5" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="6" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="-" bg="orange" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="1" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="2" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="3" bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="+" bg="orange" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="0" bg="" span="span1" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="." bg="" span="" press={(e) => handleClick(e.target.textContent)} />
+      <Button value="=" bg="orange" span="" press={(e) => handleClick(e.target.textContent)} />
```

Comment:
- [ ] I am getting an error when trying the multiply, kindly check if you are using the correct symbol 
![image](https://user-images.githubusercontent.com/62705947/234709229-d2f51b11-f066-4461-b95c-bc269e19ddbc.png)